documents/system-design-document.md

```diff
@@ -0,0 +1,82 @@
+# System Design Document for TODO List Web App
+
+## 1. Overview
+  The objective of this project is to develop a user-friendly TODO list web application that empowers users to efficiently manage their tasks. The application will provide core functionalities such as creating, reading, updating, deleting, searching, and filtering tasks within a secure environment.
+
+  For detailed project requirements, please refer to the [Product Requirements Document (PRD)](https://github.com/what-a-cool-team/docs/blob/main/prd/what-a-todo.md).
+
+## 2. Functional Requirements
+  * Users must be able to create an account or log in securely.
+  * Users can create new tasks with a name, description, and tags.
+  * Users can view a list of all thier tasks.
+  * Users can update existing tasks to change details such as name, description, status, and tags.
+  * Users can delete tasks they no longer need.
+  * Users can search tasks by name or tags.
+  * Users can filter tasks by status.
+
+## 3. Operational requirements
+  * **Scalability**: The application must be capable of scaling efficiently to accommodate increasing numbers of users and tasks.
+  * **Reliability**: The application should be designed to minimize downtime and ensure high availability to users.
+  * **Security**: 
+    - *Authentication and Authorization*: The application must implement robust mechanisms to authenticate users and authorize actions, ensuring that users can only access their own data and functionalities.
+    - *Data Protection*: User data should be protected through encryption and other security best practices, both during transmission and while stored.
+## 4. Domain Model Design
```

Comment:
Domain model is supposed to be decoupled from the physical persistence layer, but this design tries to couple them as much as possible.